rust/lance/src/index/vector/ivf.rs

```diff
@@ -767,6 +790,61 @@ pub async fn build_ivf_pq_index(
     // For now, it loads all data into memory.
     let stream = scanner.try_into_stream().await?;
 
+    let precomputed_partitions = match &ivf_params.precomputed_partitons_file {
+        Some(file) => {
+            info!("Loading precomputed partitions from file: {}", file);
+            let arrow_schema = ArrowSchema::new(vec![
+                ArrowField::new("row_id", DataType::UInt64, false),
+                ArrowField::new("partition", DataType::UInt32, false),
+            ]);
+
+            let schema = Schema::try_from(&arrow_schema)?;
+
+            let reader = FileReader::try_new_with_fragment(
+                dataset.object_store.as_ref(),
+                &dataset.base.child(DATA_DIR).child(file.as_str()),
```

Comment:
So `file` is not a full pathname but instead a child path for a file inside the data directory?  Does this file get automatically cleaned up in any way?

Reply:
the cleanup code will clean things up. there is no logic to actively clean this up. I can add one at the end of `create_index`